internal/pkg/agent/cmd/run.go

```diff
@@ -462,6 +468,19 @@ func runElasticAgent(
 		appErr <- err
 	}()
 
+	wg := new(sync.WaitGroup)
+	// Spawn the rollbacks cleanup goroutine
+	relativeHomePath, homePathErr := filepath.Rel(paths.Top(), paths.Home())
+	if homePathErr != nil {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			upgrade.PeriodicallyCleanRollbacks(ctx, l, paths.Top(), relativeHomePath, availableRollbacksSource, 10*time.Minute, 2*time.Hour)
```

Comment:
If you made the intervals configurable, you could set them to very aggressive values (similar to what we do with the fast watcher) in the existing integration tests.

I think even without an active test specifically for this, if you set the cleaner to run aggressively during the existing upgrade tests we'd regularly exercise the logic to preserve active upgrades versions.

So you could think about using the cleaner really aggressively in existing tests to try to stress test it doesn't delete versions it shouldn't, with maybe another test that the rollback overall works end to end through a normal cycle of upgrade, keep the upgrade around until TTL, cleaner removes at TTL.